### PR TITLE
Add more logging to AzureTableGrainStorage init, Add ConfiguratorValidator

### PR DESCRIPTION
**Pull request by xiazen**
- Add more logging to AzureTableGrainStorage init to improve debugging experience
- Add Configuration validator to  `AzureTableGrainStorageOptions`
- rename `DataConnectionString` to be `ConnectionString`. To be consistent with the `ConnectionString` in AzureGatewaylistprovider and AzureMembershipProvier

### CODE REVIEW

File: src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureTableStorage.cs [hunk 1]
```diff
@@ -44,16 +45,18 @@ public class AzureTableGrainStorage : IGrainStorage, IRestExceptionDecoder, ILif
 
         private const string BINARY_DATA_PROPERTY_NAME = "Data";
         private const string STRING_DATA_PROPERTY_NAME = "StringData";
-
+        private string name;
         /// <summary> Default constructor </summary>
-        public AzureTableGrainStorage(AzureTableStorageOptions options, SerializationManager serializationManager, IGrainFactory grainFactory, ITypeResolver typeResolver, ILoggerFactory loggerFactory)
+        public AzureTableGrainStorage(AzureTableStorageOptions options, SerializationManager serializationManager, IGrainFactory grainFactory, ITypeResolver typeResolver, ILoggerFactory loggerFactory, string name)
```
Comment (benjaminpetit) [inline src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureTableStorage.cs]: NIT:  I think the name should be the first parameter, it's more clear
NIT²: Could you let a space between the latest member and the ctor? You are doing the same in the validator below
NIT³: When there is a lot of param like here, I think it is better to break them in multiple lines. It's more readable, and avoid merge conflicts sometimes

File: src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureTableStorageOptions.cs [hunk 2]
```diff
@@ -45,6 +45,27 @@ public class AzureTableStorageOptions
...
+        private readonly string name;
+        public AzureTableGrainStorageOptionsValidator(AzureTableStorageOptions options, string name)
+        {
+            this.options = options;
+            this.name = name;
+        }
+
+        public void ValidateConfiguration()
+        {
+            if (string.IsNullOrEmpty(this.options.ConnectionString))
```
Comment (benjaminpetit) [inline src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureTableStorageOptions.cs]: Should we parse it to check that it is in fact an azure connection string? A simple call to `CloudStorageAccount.TryParse`maybe?

Comment (xiazen) [reply src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureTableStorageOptions.cs]: good idea

Comment (benjaminpetit) [reply src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureTableStorageOptions.cs]: I think we should also validate that the `TableName` is valid too

**Comment by xiazen** [comment]
tests passed. Pr feedback addressed

**Pull request merged**